pkg/cmd/workflow/run/run.go

```diff
@@ -0,0 +1,382 @@
+			support a workflow_dispatch 'on' trigger in order to be run in this way.
+
+			If the workflow file supports inputs, they can be specified in a few ways:
+
+			- Interactively
+			- As command line arguments specified after --
+			- As JSON, via STDIN
+			- As JSON, via --json
+    `),
+		Example: heredoc.Doc(`
+			# Have gh prompt you for what workflow you'd like to run
+			$ gh workflow run
+
+			# Run the workflow file 'triage.yml' at the remote's default branch, interactively providing inputs
+			$ gh workflow run triage.yml
+
+			# Run the workflow file 'triage.yml' at a specified ref, interactively providing inputs
+			$ gh workflow run triage.yml --ref=myBranch
+
+			# Run the workflow file 'triage.yml' with command line inputs
+			$ gh workflow run triage.yml -- --name=scully --greeting=hello
+
+			# Run the workflow file 'triage.yml' with JSON via STDIN
+			$ echo '{"name":"scully", "greeting":"hello"}' | gh workflow run triage.yml
+
+			# Run the workflow file 'triage.yml' with JSON via --json
+			$ gh workflow run triage.yml --json='{"name":"scully", "greeting":"hello"}'
+		`),
+		Args: func(cmd *cobra.Command, args []string) error {
+			if cmd.ArgsLenAtDash() == 0 && len(args[1:]) > 0 {
+				return cmdutil.FlagError{Err: fmt.Errorf("workflow argument required when passing input flags")}
+			}
+			return nil
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			// support `-R, --repo` override
+			opts.BaseRepo = f.BaseRepo
+
+			if len(args) > 0 {
+				opts.Selector = args[0]
+				opts.InputArgs = args[1:]
+			} else if !opts.IO.CanPrompt() {
+				return &cmdutil.FlagError{Err: errors.New("workflow ID, name, or filename required when not running interactively")}
+			} else {
+				opts.Prompt = true
+			}
+
+			if !opts.IO.IsStdinTTY() {
+				if opts.JSON != "" {
+					return errors.New("JSON can only be passed on one of STDIN or --json at a time")
+				}
+				jsonIn, err := ioutil.ReadAll(opts.IO.In)
+				if err != nil {
+					return errors.New("failed to read from STDIN")
+				}
+				opts.JSON = string(jsonIn)
+			}
+
+			if opts.Selector == "" {
+				if opts.JSON != "" {
+					return &cmdutil.FlagError{Err: errors.New("workflow argument required when passing JSON")}
+				}
+			} else {
+				if opts.JSON != "" && len(opts.InputArgs) > 0 {
+					return &cmdutil.FlagError{Err: errors.New("only one of JSON or input arguments can be passed at a time")}
+				}
+			}
+
+			if runF != nil {
+				return runF(opts)
+			}
+			return runRun(opts)
+		},
+	}
+	cmd.Flags().StringVarP(&opts.Ref, "ref", "r", "", "The branch or tag name which contains the version of the workflow file you'd like to run")
+	cmd.Flags().StringVar(&opts.JSON, "json", "", "Provide workflow inputs as JSON")
+
+	return cmd
+}
+
+type InputAnswer struct {
+	providedInputs map[string]string
+}
+
+func (ia *InputAnswer) WriteAnswer(name string, value interface{}) error {
```

Comment:
I might be missing something but I don't see this being used anywhere, is it possible that it was meant to be deleted?

Reply:
It's called by `survey`